opensearch_py_ml/ml_commons/rag_pipeline/rag/AIConnectorHelper.py

```diff
@@ -0,0 +1,472 @@
+
+        # Initialize IAMRoleHelper and SecretHelper
+        self.iam_helper = IAMRoleHelper(
+            region=self.region,
+            opensearch_domain_url=self.opensearch_domain_url,
+            opensearch_domain_username=self.opensearch_domain_username,
+            opensearch_domain_password=self.opensearch_domain_password,
+            aws_user_name=self.aws_user_name,
+            aws_role_name=self.aws_role_name,
+            opensearch_domain_arn=self.opensearch_domain_arn
+        )
+
+        self.secret_helper = SecretHelper(self.region)
+
+    @staticmethod
+    def get_opensearch_domain_info(region, domain_name):
+        """
+        Retrieve the OpenSearch domain endpoint and ARN based on the domain name and region.
+        """
+        try:
+            opensearch_client = boto3.client('opensearch', region_name=region)
+            response = opensearch_client.describe_domain(DomainName=domain_name)
+            domain_status = response['DomainStatus']
+            domain_endpoint = domain_status.get('Endpoint')
+            domain_arn = domain_status['ARN']
+            return domain_endpoint, domain_arn
+        except Exception as e:
+            print(f"Error retrieving OpenSearch domain info: {e}")
+            return None, None
+
+    def get_ml_auth(self, create_connector_role_name):
+        """
+        Obtain AWS4Auth credentials for ML API calls using the specified IAM role.
+        """
+        create_connector_role_arn = self.iam_helper.get_role_arn(create_connector_role_name)
+        if not create_connector_role_arn:
+            raise Exception(f"IAM role '{create_connector_role_name}' not found.")
+
+        temp_credentials = self.iam_helper.assume_role(create_connector_role_arn)
+        awsauth = AWS4Auth(
+            temp_credentials["AccessKeyId"],
+            temp_credentials["SecretAccessKey"],
+            self.region,
+            'es',
+            session_token=temp_credentials["SessionToken"],
+        )
+        return awsauth
+
+    def create_connector(self, create_connector_role_name, payload):
+        create_connector_role_arn = self.iam_helper.get_role_arn(create_connector_role_name)
+        temp_credentials = self.iam_helper.assume_role(create_connector_role_arn)
+        awsauth = AWS4Auth(
+            temp_credentials["AccessKeyId"],
+            temp_credentials["SecretAccessKey"],
+            self.region,
+            'es',
+            session_token=temp_credentials["SessionToken"],
+        )
+
+        path = '/_plugins/_ml/connectors/_create'
+        url = self.opensearch_domain_url + path
+
+        headers = {"Content-Type": "application/json"}
+
+        r = requests.post(url, auth=awsauth, json=payload, headers=headers)
+        print(r.text)
+        connector_id = json.loads(r.text)['connector_id']
+        return connector_id
+
+    def search_model_group(self, model_group_name, create_connector_role_name):
```

Comment:
where did we use this method?

Reply:
Resolved, addressed the wrapper method, and accessed existing methods directly.